test/build/custom-modules-dir/expected-build/__snowpack__/env.js

```diff
@@ -1 +1 @@
-export default {"MODE":"production","NODE_ENV":"production"};
\ No newline at end of file
+export default{MODE:"production",NODE_ENV:"production"};
```

Comment:
All the tests are minified from here... was this a mistake? I have always thought of tests as part of the documentation.

Reply:
That’s a good question. No, it wasn’t a mistake, and in all honesty I’d rather keep snapshots of the minified code considering this is what actually goes into production. I’d at least like to track minified changes.

But I hadn’t considered these being “examples,” basically. We do have [CSA templates](https://github.com/pikapkg/create-snowpack-app/), but do you think there’s room for improvement? I guess I’m asking is there something we’re losing by minifying these snapshots?

Reply:
For unit tests I typically treat them as documentation as well. But for these integration tests, I don't think there is any benefit from looking at them. They're just asserting that the builder is performing the way we would expect.

Reply:
Hello! Just saw this message. It’s great you’re thinking about using esbuild for a minifier. One consideration here is that the minified output can sometimes change in small ways between releases due to automatically-generated symbol number and ordering affecting the symbol count statistics. So doing this may cause occasional test churn for esbuild upgrades. It doesn’t happen too often but it does happen. It sounds like that’s your goal so maybe it’s fine. I’m just mentioning it because it looks like you have a lot of tests.

Reply:
> that the minified output can sometimes change in small ways between releases due to automatically-generated symbol number and ordering

That’s a great call-out. We actually commit the dependencies themselves (we don’t update or reinstall) so the only changing factor would be esbuild’s minification. And yes I think that would be good to monitor, especially with this being default Snowpack behavior for production builds. If the minification changes I would like to be flagged for review to see if it’s a significant change or not (most likely it won’t be, but still helpful to know that output is changing).

Reply:
Coming back to this, and after upgrading esbuild myself I had a change of heart. You’re right—the differences are fairly frequent. And may be deterministic per-version, but upgrading esbuild shouldn’t cause as much noise in the tests as it does.

I think for myself, I’d probably go through the pain. But I’m optimizing for **contributors**, not myself, and I know as a contributor if I started working on this and had to update _every test in the test suite_ because I upgraded one dependency, I would think “what dummy set this up? This is a horrible experience.”

I should probably go with that gut feeling and just disable optimization for the test suite because we aren’t testing esbuild’s minifier 🙂